src/mbgl/style/style_impl.cpp

```diff
@@ -255,12 +264,16 @@ Source* Style::Impl::getSource(const std::string& id) const {
     return sources.get(id);
 }
 
+bool Style::Impl::areSpritesLoaded() const {
+    return (countOfSpritesLoaded >= countOfSprites);
```

Comment:
I added an assert about this to be sure it cannot trigger issues. And it does!
It seems, when a sprite is expired, the `onSpriteLoaded()` method can be called mutliple times. Seen with the cpp unit test `Map.ExpiredSpriteSheet`. So it will be an issue.

I updated the code to store a summary of sprite object into a map in the form `std::unordered_map<std::string,bool>` in order to solve the issue.
Sadly, I still have some memory issues I need to check before the final commit.